### DISCUSSION TITLE
Drum Machine Patterns

### DISCUSSION THREAD

**Discussion by BigAlbertMusic** [hydrogen-music/hydrogen#2028] [General]
Hello,
[Drum Machine - 260 Patterns.pdf](https://github.com/user-attachments/files/16940642/Drum.Machine.-.260.Patterns.pdf)

I am currently working on encoding an open-source book of drum rhythms for sequencers "260 Drum Machine Patterns" by René Pierre Bardet.

It seems to me a good idea to allow the user to have a rhythmic base at their disposal to refine their rhythms.

I created a song by style with the patterns and breaks.

I am finishing all this and as soon as it is finished I will publish my work under GPL license.

Do you think there is any interest?

**Comment by theGreatWhiteShark** [comment]
Hey @BigAlbertMusic ,

> Do you think there is any interest?

Yes!

The main reason there are almost no Hydrogen songs or its pattern files (`.h2pattern`) online is that those are tied to a particular drumkit and its instrument order. Switching a kit to one with a different position for e.g. kick and snare would require the user to rewrite parts of the pattern.

But all this will change in version 2.0 of Hydrogen. In the current development version we are already able to seamlessly switch kits and import patterns. Your patterns would be perfect as a seed for a shared pattern data base.

> I am finishing all this and as soon as it is finished I will publish my work under GPL license.

Would you consider using a [Creative Commons](https://creativecommons.org/) license instead? GPL is tailored for software. How it applies to samples, scores etc. or if it is applicable at all is AFAIK still somewhat mysterious. (A couple of years ago I talked to an attorney of the FSFE and not even he had an opinion on the matter). Instead CC-BY-SA is somewhat similar to GPL but gives artists a lot more certainty about they rights and obligations regarding your work.

**Comment by BigAlbertMusic** [comment]
Thank you for your feedback.
It is the reason I encoded with the default drum kit GM Rock even I prefer AVL Black Pearl.
Of course no problem to publish in CC-BY-SA, it is already the licence wich I share my music ;-)
Here is my work.
[260 Drum Machine Patterns from RP Bardet.zip](https://github.com/user-attachments/files/16950160/260.Drum.Machine.Patterns.from.RP.Bardet.zip)

**Reply by theGreatWhiteShark** [reply]
Thanks a lot!

**Reply by theGreatWhiteShark** [reply]
I would like to share your songs and the contained patterns via our repo https://github.com/hydrogen-music/Song-and-pattern-repository. But currently their `<author>` and `<license>` elements are not set.

`<author>` is required if you want an attribution (the `-BY` part in an CC license). Except you want to go with `CC0` and allow users to use them however they like.

`<license>` is mandatory. Else we could get in legal issues.

I can also fill those elements for you if you tell me what to fill them with.

**Reply by BigAlbertMusic** [reply]
You can credit Albert Phelipot and license is CC-BY-SA. Thank you for filling this.

I am working on another piece much professional because I encode my drum patterns for my tracks. Let you know when ready in few months.
It would be very cool to have songs independant of the drumkit.
Thank you.

**Reply by theGreatWhiteShark** [reply]
>  Let you know when ready in few months.

You could subscribe to our mailing list https://sourceforge.net/projects/hydrogen/lists/hydrogen-devel or watch this repo with notifications set to releases only. Watch out for `v2.0` (alpha/beta) releases. The `1.2.X` line won't have drumkit-indepedent patterns.

**Reply by forrestkuen** [reply]
thank you! how can i download these patterns?
https://github.com/hydrogen-music/Song-and-pattern-repository

**Reply by theGreatWhiteShark** [reply]
> how can i download these patterns?

For now they are not included in the online library accessible through Hydrogen itself.

But you can just click a song in https://github.com/hydrogen-music/Song-and-pattern-repository/tree/main/Songs/260_Drum_Machine_Patterns_from_RP_Bardet and use the "Download raw file" button in the header section.

**Comment by BigAlbertMusic** [comment]
Happy to contribute to this great software I use in many of my productions.
You can ear Hydrogen Drum Machine here by example : https://youtu.be/78Y2blVrkeQ with the famous AVL Black Pearl drumkit.
Thank you to the dev team.
